broker/src/main/java/io/zeebe/broker/clustering/atomix/storage/snapshot/AtomixSnapshotStorage.java

```diff
@@ -23,32 +23,37 @@
 import java.util.Set;
 import java.util.concurrent.CopyOnWriteArraySet;
 import java.util.stream.Stream;
+import org.agrona.IoUtil;
 import org.slf4j.Logger;
 
 public final class AtomixSnapshotStorage implements SnapshotStorage, SnapshotListener {
 
   private static final Logger LOGGER = new ZbLogger(AtomixSnapshotStorage.class);
 
   private final Path runtimeDirectory;
+  private final Path pendingDirectory;
   private final AtomixRecordEntrySupplier entrySupplier;
   private final SnapshotStore store;
   private final Set<SnapshotDeletionListener> deletionListeners;
   private final SnapshotMetrics metrics;
 
   public AtomixSnapshotStorage(
       final Path runtimeDirectory,
+      final Path pendingDirectory,
       final SnapshotStore store,
       final AtomixRecordEntrySupplier entrySupplier,
       final SnapshotMetrics metrics) {
     this.runtimeDirectory = runtimeDirectory;
+    this.pendingDirectory = pendingDirectory;
     this.entrySupplier = entrySupplier;
     this.store = store;
     this.metrics = metrics;
 
     this.deletionListeners = new CopyOnWriteArraySet<>();
-    this.store.addListener(this);
 
+    IoUtil.ensureDirectoryExists(pendingDirectory.toFile(), "push replication pending directory");
```

Comment:
We can use `FileUtil:ensureDirectoryExists` which you added in another PR to be consistent.

Reply:
As soon as the other PR is merged :+1: